src/AggregateFunctions/AggregateFunctionQuantile.h

```diff
@@ -125,7 +125,7 @@ class AggregateFunctionQuantile final
         if constexpr (std::is_same_v<Data, QuantileTiming<Value>>)
         {
             /// QuantileTiming only supports unsigned integers. Too large values are also meaningless.
-            if (isNaN(value) || value > std::numeric_limits<Int64>::max() || value < 0)
+            if (isNaN(value) || value > std::numeric_limits<Value>::max() || value < 0)
```

Comment:
`value` is calculated from a `const ColVecType &` which is equivalent to `const ColumnVectorOrDecimal<Value> &`. This means that `value` is of type `Value`. So `value > std::numeric_limits<Value>::max()` doesn't make sense (it's always false). This needs to stay `std::numeric_limits<Int64>`. (I also think that the test failures are due to that)

The compiler complains on MacOS as follows:

```
src/AggregateFunctions/AggregateFunctionQuantile.h:128:41: error: implicit conversion from 'type' (aka 'long long') to 'float' changes value from 9223372036854775807 to 9223372036854775808 [-Werror,-Wimplicit-const-int-float-conversion]
            if (isNaN(value) || value > std::numeric_limits<Int64>::max() || value < 0)                                                                                  
                                      ~ ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

Basically, the right side is promoted to `float` and that reduces precision for big values. Interestingly, this doesn't happen on Linux - it is not really clear to me why.

Imho, the "best" solution would be to suppress the warning
```
#pragma clang diagnostic push
#pragma clang diagnostic ignored "-Wimplicit-const-int-float-conversion"
...
#pragma clang diagnostic pop
```

(if you want to go fancy, you can restrict this to only MacOS (use `_Pragma` and `OS_DARWIN`) for that.)